homeassistant/components/iaqualink/__init__.py

```diff
@@ -101,24 +103,36 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     aqualink = AqualinkClient(username, password, session)
     try:
         await aqualink.login()
-    except AqualinkLoginException as login_exception:
+    except AqualinkServiceException as login_exception:
         _LOGGER.error("Failed to login: %s", login_exception)
         return False
     except (
         asyncio.TimeoutError,
         aiohttp.client_exceptions.ClientConnectorError,
     ) as aio_exception:
-        _LOGGER.warning("Exception raised while attempting to login: %s", aio_exception)
-        raise ConfigEntryNotReady from aio_exception
+        raise ConfigEntryNotReady(
+            f"Errr while attempting login: {aio_exception}"
+        ) from aio_exception
+
+    try:
+        systems = await aqualink.get_systems()
+    except AqualinkServiceException as svc_exception:
+        raise ConfigEntryNotReady(
+            f"Error while attempting to retrieve systems list: {svc_exception}"
```

Comment:
To test this, setup the integration, and mock `get_systems` throwing `AqualinkServiceException`.  Then assert that the state of the config entry is SETUP_RETRY

Reply:
Should already be covered by test_setup_systems_exception()?